code/game/objects/items/GriffeningPorted.dm

```diff
@@ -0,0 +1,805 @@
+	cards -= choice
+	user.visible_message("[user] draws a card from the deck.", "<span class='notice'>You draw a card from the deck.</span>")
+	update_icon()
+	. = ..()
+
+/obj/item/griffeningdeck/Initialize()
+	. = ..()
+	if(length(GLOB.CardCreatures) == 0) //If this is full and the other lists are empty something got really messed up
+		GLOB.CardCreatures = subtypesof(/obj/item/griffeningdeck/cardhand/single/creature)
+		GLOB.CardEffects = subtypesof(/obj/item/griffeningdeck/cardhand/single/effect)
+		GLOB.CardEquipment = subtypesof(/obj/item/griffeningdeck/cardhand/single/Equipment)
+		GLOB.CardAreas = subtypesof(/obj/item/griffeningdeck/cardhand/single/Area)
+	if(IsBooster) //Will it generate stuff?
+		for(var/i = 0; i <= CardsGenerated; i++)
+		var/randomnumber = rand(1, 10)
+		switch(randomnumber)
+		if(1 to 3)
+			generate_creature()
+		if(4 to 6)
+			generate_effect()
+		if(7 to 9)
+			generate_equipment()
+		if(10)
+			generate_area()
+
+/obj/item/griffeningdeck/interact(mob/user)
+	var/dat = "You have:<BR>"
+	for(var/t in cards)
+		dat += "<A href='?src=\ref[src];pick=[t]'>A [t].</A><BR>"
+	dat += "Which card will you remove next?"
+	var/datum/browser/popup = new(user, "cardhand", "Hand of Cards", 400, 240)
+	popup.set_title_image(user.browse_rsc_icon(src.icon, src.icon_state))
+	popup.set_content(dat)
+	popup.open()
+	. = ..()
+
+/obj/item/griffeningdeck/Topic(href, href_list)
+	if(. = ..())
+		return
+	if(usr.stat || !ishuman(usr) || !usr.canmove)
+		return
+	var/mob/living/carbon/human/cardUser = usr
+	if(href_list["pick"])
+		if (cardUser.is_holding(src))
+			var/choice = href_list["pick"]
+			var/N = new choice(src.loc)
+			src.cards -= choice
+			cardUser.put_in_hands(N)
+			cardUser.visible_message("<span class='notice'>[cardUser] draws a card from [cardUser.p_their()] hand.</span>", "<span class='notice'>You take the card from your hand.</span>")
+			interact(cardUser)
+			if(src.cards.len < 3)
+				src.icon_state = "[deckstyle]_hand2"
+			else if(src.cards.len < 4)
+				src.icon_state = "[deckstyle]_hand3"
+			else if(src.cards.len < 5)
+				src.icon_state = "[deckstyle]_hand4"
+			if(src.cards.len == 1)
+				var/A = src.cards[1]
+				qdel(src)
+				cardUser.put_in_hands(A)
+				to_chat(cardUser, "<span class='notice'>You also take [cards[1]] and hold it.</span>")
+				cardUser << browse(null, "window=cardhand")
+			return
+
+
+/obj/item/griffeningdeck/attackby(obj/item/I, mob/living/user, params)
+	if(istype(I, /obj/item/griffeningdeck/cardhand/single/))
+		if(!user.temporarilyRemoveItemFromInventory(I))
+			to_chat(user, "<span class='warning'>The card is stuck to your hand, you can't add it to the deck!</span>")
+			return
+		cards += I
+		user.visible_message("[user] adds a card to the bottom of the deck.","<span class='notice'>You add the card to the bottom of the deck.</span>")
+		qdel(I)
+		update_icon()
+
+/obj/item/griffeningdeck/bigbooster
+	IsBooster = TRUE
+	CardsGenerated = 30
+
+/obj/item/griffeningdeck/cardhand
```

Comment:
Why are hands a child of decks when they literally don't inherit ANY behavior from the deck?